run_trainer.sh

```diff
@@ -0,0 +1,150 @@
+TEST_ARTIFACTS_PATH="${ARTIFACTS_PATH}/$CONDA_ENV_NAME/t_${TESTNAME}/${JOB_ID}"
+mkdir -p "$TEST_ARTIFACTS_PATH"
+
+echo "NEURON RUN DIRECTORY ${TEST_ARTIFACTS_PATH} ${SLURM_JOB_ID} ${RESUME_JOB_ID}"
+
+NEURON_DUMP_PATH=${TEST_ARTIFACTS_PATH}/neuron_dump
+HLO_DUMP_PATH=${TEST_ARTIFACTS_PATH}/hlo_dump
+
+export XLA_PYTHON_CLIENT_PREALLOCATE=false
+export XLA_PYTHON_CLIENT_ALLOCATOR=platform
+
+# export JAX_PLATFORMS=cpu
+
+#Perf Tuning Guideline here : https://github.com/NVIDIA/JAX-Toolbox/blob/main/rosetta/docs/PGLE.md
+#export XLA_FLAGS="--xla_dump_hlo_as_text --xla_dump_to=${HLO_DUMP_PATH} --xla_dump_hlo_pass_re='.*' --xla_dump_hlo_as_proto"
+:'
+if [[ $GPU_RUN_TYPE == "profile" ]]; then
+   echo "Executing GPU Profile Run"
+   XLA_FLAGS="--xla_gpu_enable_latency_hiding_scheduler=false --xla_gpu_disable_async_collectives=allreduce,allgather,reducescatter,collectivebroadcast,alltoall,collectivepermute"
+elif [[ $GPU_RUN_TYPE == "perf" ]]; then
+   echo "Executing GPU Performance Run"
+   if [[ -z "${GPU_PROFILE}" ]]; then
+       echo "ERROR : Can not run GPU Performance Run without a profile"
+       exit 1
+   fi
+   XLA_FLAGS="--xla_gpu_enable_latency_hiding_scheduler=true
+     --xla_gpu_enable_triton_gemm=false
+     --xla_gpu_graph_level=0
+     --xla_gpu_all_reduce_combine_threshold_bytes=1073741824
+     --xla_gpu_all_gather_combine_threshold_bytes=1073741824
+     --xla_gpu_reduce_scatter_combine_threshold_bytes=1073741824
+     --xla_gpu_enable_pipelined_all_gather=true
+     --xla_gpu_enable_pipelined_reduce_scatter=true
+     --xla_gpu_enable_pipelined_all_reduce=true
+     --xla_gpu_enable_while_loop_double_buffering=true
+     --xla_gpu_enable_all_gather_combine_by_dim=false
+     --xla_gpu_enable_reduce_scatter_combine_by_dim=false
+     --xla_disable_hlo_passes=rematerialization
+     --xla_gpu_pgle_profile_file_or_directory_path=${GPU_PROFILE}
+     ${XLA_FLAGS}" 
+else
+   #perf run with no profile
+   echo "Executing GPU Run with no profile"
+   XLA_FLAGS="--xla_gpu_enable_latency_hiding_scheduler=true --xla_gpu_enable_while_loop_double_buffering=true --xla_gpu_enable_pipelined_all_gather=true --xla_gpu_enable_pipelined_reduce_scatter=true --xla_gpu_enable_pipelined_all_reduce=true --xla_gpu_multi_streamed_windowed_einsum=true --xla_gpu_enable_custom_fusions=true --xla_gpu_enable_address_computation_fusion=true ${XLA_FLAGS}"
+fi
+'
+
+###export NCCL_DEBUG=INFO
+###export NCCL_DEBUG_SUBSYS=COLL
+
+#HAH quick fix
+export XLA_FLAGS="--xla_dump_hlo_as_text --xla_dump_to=${HLO_DUMP_PATH} --xla_dump_hlo_pass_re='.*' --xla_dump_hlo_as_proto --xla_gpu_enable_latency_hiding_scheduler=true --xla_gpu_enable_while_loop_double_buffering=true --xla_gpu_enable_pipelined_all_gather=true --xla_gpu_enable_pipelined_reduce_scatter=true --xla_gpu_enable_pipelined_all_reduce=true --xla_gpu_multi_streamed_windowed_einsum=true --xla_gpu_enable_custom_fusions=true" # --xla_gpu_enable_address_computation_fusion=true"
```

Comment:
why are GPU flags in TRN runs?

Reply:
This is just the checkpointing script.
@HahTK I need a walkthrough of this script to actually clean it up. I havent used this to launch Trn jobs, I was using Apoorv's launch script.